tests/Fractions.Tests/FractionSpecs/ToDataType/Method_ToDataType.cs

```diff
@@ -219,4 +351,442 @@ public class When_converting_NegativeInfinity : Spec {
+        Invoking(() => UIntPtr.CreateChecked(fraction)).Should().Throw<OverflowException>();
+        short.CreateChecked(fraction).Should().Be(-1);
+        Invoking(() => ushort.CreateChecked(fraction)).Should().Throw<OverflowException>();
+        Invoking(() => byte.CreateChecked(fraction)).Should().Throw<OverflowException>();
+        sbyte.CreateChecked(fraction).Should().Be(-1);
+    }
+
+    [Test]
+    public void CreateSaturating_should_return_the_expected_value() {
+        var fraction = new Fraction(-3, 2);
+
+        double.CreateSaturating(fraction).Should().Be(-1.5);
+        Complex.CreateSaturating(fraction).Should().Be(new Complex(-1.5, 0));
+        decimal.CreateSaturating(fraction).Should().Be(-1.5m);
+        float.CreateSaturating(fraction).Should().Be(-1.5f);
+        Half.CreateSaturating(fraction).Should().Be((Half)(-1.5));
+        BigInteger.CreateSaturating(fraction).Should().Be(-1);
+        Int128.CreateSaturating(fraction).Should().Be(-1);
+        UInt128.CreateSaturating(fraction).Should().Be(0);
+        long.CreateSaturating(fraction).Should().Be(-1);
+        ulong.CreateSaturating(fraction).Should().Be(0);
+        int.CreateSaturating(fraction).Should().Be(-1);
+        uint.CreateSaturating(fraction).Should().Be(0);
+        nint.CreateSaturating(fraction).Should().Be(-1);
+        UIntPtr.CreateSaturating(fraction).Should().Be(0);
+        short.CreateSaturating(fraction).Should().Be(-1);
+        ushort.CreateSaturating(fraction).Should().Be(0);
+        byte.CreateSaturating(fraction).Should().Be(0);
+        sbyte.CreateSaturating(fraction).Should().Be(-1);
+    }
+
+    [Test]
+    public void CreateTruncating_should_return_the_expected_value() {
+        var fraction = new Fraction(-3, 2);
+
+        double.CreateTruncating(fraction).Should().Be(-1.5);
+        Complex.CreateTruncating(fraction).Should().Be(new Complex(-1.5, 0));
+        decimal.CreateTruncating(fraction).Should().Be(-1.5m);
+        float.CreateTruncating(fraction).Should().Be(-1.5f);
+        Half.CreateTruncating(fraction).Should().Be((Half)(-1.5));
+        BigInteger.CreateTruncating(fraction).Should().Be(-1);
+        Int128.CreateTruncating(fraction).Should().Be(-1);
+        UInt128.CreateTruncating(fraction).Should().Be(new UInt128(18446744073709551615, 18446744073709551615));
+        long.CreateTruncating(fraction).Should().Be(-1);
+        ulong.CreateTruncating(fraction).Should().Be(18446744073709551615UL);
+        int.CreateTruncating(fraction).Should().Be(-1);
+        uint.CreateTruncating(fraction).Should().Be(4294967295u);
+        nint.CreateTruncating(fraction).Should().Be(-1);
+        UIntPtr.CreateTruncating(fraction).Should().Be(new UIntPtr(18446744073709551615));
+        short.CreateTruncating(fraction).Should().Be(-1);
+        ushort.CreateTruncating(fraction).Should().Be(65535);
+        byte.CreateTruncating(fraction).Should().Be(255);
+        sbyte.CreateTruncating(fraction).Should().Be(-1);
+    }

```

Comment:
This is another contentious point- currently all integer based types (including the unsigned versions) are created using an intermediate conversion to `BigInteger`:
- `TOther.CreateSaturating((BigInteger)fraction)  // where TOther might be the "int" type`
- `TOther.CreateTruncating((BigInteger)fraction) `

The `BigInteger` has separate implementation for each of the `TryConvertTo..` versions, with the truncating one, doing the overflowing as demonstrated by the second set of tests.
However, as I was looking at the sources for the `decimal`/`double` types, it seems they have just one implementation for the `TryConvertToSaturating` and `TryConvertToTruncating` (with the latter calling the former)..

I assume, whoever wants to create unsigned numbers from fractions knows what he's doing.. 🤷